src/AggregateFunctions/AggregateFunctionQuantile.h

```diff
@@ -87,7 +91,36 @@ class AggregateFunctionQuantile final
         if (!returns_many && levels.size() > 1)
             throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Aggregate function {} requires one level parameter or less", getName());
 
-        if constexpr (has_accuracy_parameter)
+        if constexpr (is_quantile_ddsketch)
+        {
+            if (params.empty())
+                throw Exception(
+                    ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Aggregate function {} requires at least one param", getName());
+
+            const auto & relative_accuracy_field = params[0];
+            if (!isFloat64FieldType(relative_accuracy_field.getType()))
```

Comment:
I think you can actually use `argument_types_` to find out if the param is a float64 like so:
```
            if(!WhichDataType(argument_types_[0]).isFloat64())
```
If that works, you possibly won't need to add this new function `isFloat64FieldType()`.

Reply:
We want to get value and type both and I saw `isInt64OrUInt64FieldType` being used elsewhere in the codebase and used it for simplicity. I don't see any advantage of this over it.